### DISCUSSION TITLE
A ScreenHandler method for processing the logic of inventory slots changes.

### DISCUSSION THREAD

**Discussion by Intezium** [FabricMC/fabric-api#2404] [Mod Dev Support] [answered]
I have a ScreenHandler. The onSlotClick method processes the logic of my BlockEntity container. 
Did I choose the right method to handle the logic of changing the container inventory slots?

**Comment by warjort** [comment, marked as answer]
You don't really explain what you are doing. 
"Processes the logic of my BlockEntity container" is too vague.

Usually, you handle different behaviours by creating special Slot implementations.
e.g. Look at BrewingStandScreenHandler that has FuelSlot, IngredientSlot and PotionSlots inner classes.

Find a vanilla BlockEntity that does something similar to what you are trying to do and see what it does with its slots.

**Reply by Intezium** [reply]
Sorry for the vague wording. I do some actions with NBT in items in container inventory under certain conditions availability items in container inventory slots.
